### PR TITLE
filogic: update ZyXEL EX5700 DTS, add missing LED configuration

### PR DESCRIPTION
**Pull request by nlzy**

### CODE REVIEW

**Comment by hauke** [comment]
@bmork Please have a look

**Comment by bmork** [comment]
My immediate thought was that this looks fine.  I didn't include any LED config simply because we didn't have that feature when the dts was originally submitted.

Looking closer now, I see that this completely disables the second (amber) port LED.  The remaining phy LED outputs are not connected, so disabling them makes sense.  But leaving one LED always dark is maybe not the best use of the hardware?  These ports aren't directly comparable to the 3 other ports anyway,  supporting 4 speeds.  We could at least make some distinction between 2500 and 1000/100/10.  

FWIW, whatever we do I do not necessarily think we should follow OEM as a guide here.  Their LED config is weird.  It is
```
 maxlinear,led-reg = <0x3c0 0x330 0x00 0x00>;
```
which decodes to green LED on 2500 + 1000 with pulse on traffic and amber LED on 100 + 10 with pulse on traffic

**Review by bmork — COMMENTED** [review]
This makes sense only if the DTS is used by U-Boot, isn't that so?  Is the plan to build a replacement bootloader using this DTS?  My focus when adding this device was to live with the quirks imposed by the existing bootloader if possible.

The original bootloader on these devices has some "interesting" vendor code. It tries to load a "factoryparams" ubi volume:
```
 Loading Environment from NVRAM... OK
In:    serial@11002000
Out:   serial@11002000
Err:   serial@11002000
Model: ex5700-t0
tpl_env not found in U-Boot DTB [-1]
tpl-version not found in U-Boot DTB [-1]
mtd: partition "ubi" extends beyond the end of device "nmbm0" -- size truncated to 0x1da80000
ubi0: attaching mtd6
ubi0: scanning is finished
ubi0: attached mtd6 (name "ubi", size 474 MiB)
ubi0: PEB size: 262144 bytes (256 KiB), LEB size: 253952 bytes
ubi0: min./max. I/O unit sizes: 4096/4096, sub-page size 4096
ubi0: VID header offset: 4096 (aligned 4096), data offset: 8192
ubi0: good PEBs: 1898, bad PEBs: 0, corrupted PEBs: 0
ubi0: user volume: 11, internal volumes: 1, max. volumes count: 128
ubi0: max/mean erase counter: 124/68, WL threshold: 4096, image sequence number: 1659616596
ubi0: available PEBs: 2, total reserved PEBs: 1896, PEBs reserved for bad PEB handling: 38
Volume factoryparams not found!
sah_rip_find_dtb_in_ubi: Unable to read ubi volume from ubi:factoryparams **
parse_dtb: Failed to read the RIP DTB
board_late_init: sah_rip_init failed.
board_late_init: can't find WAN_ADDR in RIP
Saving Environment to NVRAM... OK
```

This volume does not exist on any of the original samples I've seen.  All factory data, except the wifi eeproms, is stored in the "RIP" ubi volume.  This is used directly by the OEM userspace.

Because of the failure above, the original bootloader ends up setting ethaddr to a random address.  But if we copy the "RIP" volume to "factoryparams" we get the proper wan mac in ethaddr:
```
Loading Environment from NVRAM... OK
In:    serial@11002000
Out:   serial@11002000
Err:   serial@11002000
Model: ex5700-t0
tpl_env not found in U-Boot DTB [-1]
tpl-version not found in U-Boot DTB [-1]
mtd: partition "ubi" extends beyond the end of device "nmbm0" -- size truncated to 0x1da80000
ubi0: attaching mtd6
ubi0: scanning is finished
ubi0: attached mtd6 (name "ubi", size 474 MiB)
ubi0: PEB size: 262144 bytes (256 KiB), LEB size: 253952 bytes
ubi0: min./max. I/O unit sizes: 4096/4096, sub-page size 4096
ubi0: VID header offset: 4096 (aligned 4096), data offset: 8192
ubi0: good PEBs: 1898, bad PEBs: 0, corrupted PEBs: 0
ubi0: user volume: 10, internal volumes: 1, max. volumes count: 128
ubi0: max/mean erase counter: 124/68, WL threshold: 4096, image sequence number: 1659616596
ubi0: available PEBs: 41, total reserved PEBs: 1857, PEBs reserved for bad PEB handling: 38
Read 40 bytes from volume factoryparams to 0000000077bfdaf8
Read 5124 bytes from volume factoryparams to 0000000077f73370
Saving Environment to NVRAM... OK

=> printenv ethaddr
ethaddr=f0:87:56:cc:52:30
=> rip list          
SERIAL_NUMBER
WAN_ADDR
MANUFACTURER
MANUFACTURER_OUI
MANUFACTURER_URL
MODELNAME
PRODUCT_CLASS
HARDWARE_VERSION
BOOTLOADER_VERSION
CLIENT_CERTIFICATE
PRIVATE_KEY
WLAN_SSID
WLAN_KEY
SSH_PUB_KEY
SSH_ROOT_PWD
ADMIN_PWD
UNIQUE_PRIVATE_KEY
GLOBAL_PRIVATE_KEY
USERFS_KEY
=> rip read WAN_ADDR
WAN_ADDR:
        index: 2
        secure: 0
        is_encrypted: 0
        is_signed: 0
        length: 6
        data=f0:87:56:cc:52:30
```

I considered making OpenWrt automatically create such a copy, but dropped it. It's not our task to fix OEM bootloader issues.  But I do see the problem with a random ethaddr.

Maybe we should copy RIP to factoryparams during sysupgrade if there is no such volume already?

**Comment by nlzy** [comment]
@bmork 

> But leaving one LED always dark is maybe not the best use of the hardware?

Totally agree. This configuration is not the best, it's just better than current configuration, and keep consistent with other port LED.

I think it's fine, but if there is any better solution, I'd be happy to take.

> This makes sense only if the DTS is used by U-Boot, isn't that so?

I'm not familiar with u-boot. Seems u-boot will read the DTB of kernel, and add 2 nodes named 'local-mac-address' and 'mac-address', then kernel will follow those settings. 

BTW, I'm not using OEM u-boot, it always dropped into failsafe mode after using `reboot' command on openwrt. So I replaced with other u-boot. More precisely, I'm using

https://github.com/hanwckf/bl-mt798x

Similar processing found in origin u-boot sources, and it was documented. I think improve compatibility with u-boot and other derivatives is not a bad idea.

**Comment by bmork** [comment]
tl;dr; I propose` <0x3f0 0x330 0 0>`

Had a chance to test the LEDs now, but forgot to bring a 2500 capable adapter so I didn't test that.  There's no reason to believe it doesn't work as documented though.

AFAICS, the three uppermost ports (lan1, lan2 and lan3) using the MT7531 builtin PHYs are not configurable and fixed to
- 1000: green with blink on TX/RX
- 100: green+yellow with blink on TX/RX
- 10: green+yellow with blink on TX/RX

I believe we should try to match that on the gpy211 ports (lan4 and wan).  This doesn't help for 2500 of course, but IMHO the least unexpected is using the same pattern as for 1000. I.e
- 2500: green with blink on TX/RX
- 1000: green with blink on TX/RX
- 100: green+yellow with blink on TX/RX
- 10: green+yellow with blink on TX/RX

The green LED is connected  gpy211 LED pin 0 and the yellow LED is connected to gpy211 LED pin 1. Pin 2 and 3 are not connected. So the gpy211 register values matching that is
`<0x3f0 0x330 0 0>
`
It would sort of be nice to have a separate 2500 indication, but I believe the combinations we have left will be either confusing or annoying (or both). By "annoying" I'm thinking of the "fast blink" option gpy211 supports.  This looks like flickering.  I wouldn't want that.....


BTW, in case you don't know:  It is pretty easy to test out the different patterns using the mdio-tools package.  You can read and change the gpy211 registers on the fly (this is for lan4 - use 6 instead of 5 for wan):
```
root@OpenWrt:/# mdio mdio-bus mmd 5:30 raw 1
0x03f0
root@OpenWrt:/# mdio mdio-bus mmd 5:30 raw 2
0x0000
root@OpenWrt:/# mdio mdio-bus mmd 5:30 raw 3
0x0000
root@OpenWrt:/# mdio mdio-bus mmd 5:30 raw 4
0x0000
root@OpenWrt:/# mdio mdio-bus mmd 5:30 raw 1 0x0310
```

**Comment by bmork** [comment]
> I'm not using OEM u-boot,

That's fine, but it is a different device IMHO. That's the reason I included "Telenor" in the description.  What we currently support is that specific bootloader, which has some "interesting" quirks. You should create device variants for any significantly different bootloader. Those will probably not have the in-memory u-boot environment etc. 

Personally I think we should be careful with recommending bootloader updates to regular end users.  That's bricking land.  Most OEM bootloaders are buggy of course, but I'd rather live with a few supportable bugs than having to explain end users that "sorry, there is nothing more we can do after you misread our instructions.  It's a permanent brick"

**Comment by GoetzGoerisch** [comment]
Thank you for your PR.

Just a heads up, we had a recent [treewide change from ZyXEL to Zyxel](https://github.com/openwrt/openwrt/commit/3774f3272ee976b922306a881b6eb603e79f81ed).
Please adjust your commit message accordingly. Thank you!

**Comment by nlzy** [comment]
I force push with new commit

>  I propose <0x3f0 0x330 0 0>

Thanks for your propose. I have already changed to this.

> but forgot to bring a 2500 capable adapter so I didn't test that

I have tested with 2500 link speed. LED works as expected.

> You should create device variants for any significantly different bootloader.

According to your opinion, I removed the commit that added ethernet1 alias, and changed the PR title.

> Please adjust your commit message accordingly

Done
